data/data-pipeline/data_pipeline/comparison_tool/src/tract_comparison__template.ipynb

```diff
@@ -0,0 +1,379 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "618d2bac-8800-4fd7-a75d-f4abaae30923",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#papermill_description=Tract_stats\n",
+    "tract_level_by_identification = pd.concat(\n",
+    "    [\n",
+    "        utils.get_demo_series(\n",
+    "            COMPARATOR_COLUMN, joined_frame, ADDITIONAL_DEMO_COLUMNS + DEMOGRAPHIC_COLUMNS\n",
+    "        ),\n",
+    "        utils.get_demo_series(\n",
+    "            SCORE_COLUMN, joined_frame, ADDITIONAL_DEMO_COLUMNS + DEMOGRAPHIC_COLUMNS\n",
+    "        ),\n",
+    "    ],\n",
+    "    axis=1,\n",
+    ")\n",
+    "\n",
+    "tract_level_by_identification"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f37bfa5-5e9e-46d1-97af-8d0e7d53c80b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(11, 11))\n",
+    "sns.barplot(\n",
+    "    y=\"Variable\",\n",
+    "    x=\"Avg in tracts\",\n",
+    "    hue=\"Definition\",\n",
+    "    data=tract_level_by_identification.sort_values(by=COMPARATOR_COLUMN, ascending=False)\n",
+    "    .stack()\n",
+    "    .reset_index()\n",
+    "    .rename(\n",
+    "        columns={\"level_0\": \"Variable\", \"level_1\": \"Definition\", 0: \"Avg in tracts\"}\n",
+    "    ),\n",
+    "    palette=\"Blues\",\n",
+    ")\n",
+    "plt.xlim(0, 1)\n",
+    "plt.title(\"Tract level averages by identification strategy\")\n",
+    "plt.savefig(os.path.join(OUTPUT_DATA_PATH, \"tract_lvl_avg.jpg\"), bbox_inches='tight')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "713f2a04-1b2a-472c-b036-454cd5c9a495",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#papermill_description=Tract_stats_grouped\n",
+    "tract_level_by_grouping = utils.get_tract_level_grouping(\n",
+    "    joined_frame,\n",
+    "    SCORE_COLUMN,\n",
+    "    COMPARATOR_COLUMN,\n",
+    "    ADDITIONAL_DEMO_COLUMNS + DEMOGRAPHIC_COLUMNS,\n",
+    ")\n",
+    "\n",
+    "tract_level_by_grouping_formatted = utils.format_multi_index_for_excel(\n",
+    "    tract_level_by_grouping\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d166f53-4242-46f1-aedf-51dac517de94",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tract_level_by_grouping_formatted"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "372ac341-b03b-404a-9462-5087b88579b7",
+   "metadata": {},
+   "source": [
+    "## Population-weighted stats"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d172e3c8-60ab-44f0-8034-dda69f1146da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#papermill_description=Population_stats\n",
+    "population_weighted_stats = pd.concat(\n",
+    "    [\n",
+    "        utils.construct_weighted_statistics(\n",
+    "            joined_frame,\n",
```

Comment:
nit: we don't document this anywhere, but I think @esfoobar-usds and I both prefer named arguments wherever possible. In past codebases I've actually implemented linters to force named arguments on every single function/method invocation. 

It just makes the invocation much easier to read -- on my first glance at this, I have no idea what these four parameters are supposed to be doing –– and therefore much easier to debug.

Reply:
👍  makes sense to me!